qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -168,19 +167,31 @@ def _compute_outcome_probabilities(self, circuits: List[QuantumCircuit]):
 
 
 class MockIQBackend(FakeOpenPulse2Q):
-    """An abstract backend for testing that can mock IQ data."""
+    """A mock backend for testing with IQ data."""
 
     def __init__(
         self,
-        iq_cluster_centers: Tuple[float, float, float, float] = (1.0, 1.0, -1.0, -1.0),
-        iq_cluster_width: float = 1.0,
+        experiment_helper: MockIQExperimentHelper,
         rng_seed: int = 0,
+        iq_cluster_centers: Optional[List[Tuple[Tuple[float, float], Tuple[float, float]]]] = None,
+        iq_cluster_width: Optional[List[float]] = None,
     ):
         """
         Initialize the backend.
+        Args:
+            experiment_helper(MockIQExperimentHelper): Experiment helper class that contains
+            'compute_probabilities' function and 'iq_phase' function for the backend to execute.
+            rng_seed(int): The random seed value.
+            iq_cluster_centers(Optional[List]): A list of tuples containing the clusters' centers in the
+            IQ plane.
+            There are different centers for different logical values of the qubit.
+            iq_cluster_width(Optional[List]): A list of standard deviation values for the sampling of
+            each qubit.
         """
-        self._iq_cluster_centers = iq_cluster_centers
-        self._iq_cluster_width = iq_cluster_width
+
+        self._iq_cluster_centers = iq_cluster_centers or [((-1.0, -1.0), (1.0, 1.0))]
+        self._iq_cluster_width = iq_cluster_width or [1.0 for _ in self._iq_cluster_centers]
+        self.experiment_helper = experiment_helper
```

Comment:
Is there a reason that the variable's name is `experiment_helper` and not `_experiment_helper`?

Reply:
I wanted the experiment helper to be public so if someone can use the same instance of backend for different experiments (or different configuration). Because there is no additional logic I need to implement for the `setter` me and @eliarbel thought its ok to make the attribute public.
Example 1:
For the same backend (cluster centers and widths) I want to run Ramsey experiment and then Fine amplitude. All I need is to change is the `experiment_helper`.

Example 2:
I want to run Ramsey with two different `T2` value, I will do it by changing the experiment helper.